Soomsil-USaint/Data/Client/RuSaint/GradeClient.swift

```diff
@@ -4,4 +4,253 @@
+                               overallRank: newSemester.overallRank,
+                               overallStudentCount: newSemester.overallStudentCount,
+                               lectures: newSemester.lectures ?? nil,
+                               in: context)
+                
+                try context.save()
+            },
+            deleteAllSemesterGrades: {
+                let context = coreDataStack.taskContext()
+                let deleteRequest = NSBatchDeleteRequest(fetchRequest: CDSemester.fetchRequest())
+                
+                try context.execute(deleteRequest)
+                try context.save()
+            },
+            deleteGrades: { year,semester in
+                let context = coreDataStack.taskContext()
+                let fetchRequest: NSFetchRequest<CDSemester> = CDSemester.fetchRequest()
+                fetchRequest.predicate = NSCompoundPredicate(andPredicateWithSubpredicates: [
+                    NSPredicate(format: "year == %d", year),
+                    NSPredicate(format: "semester == %@", semester)
+                ])
+                
+                let fetchedEntity = try context.fetch(fetchRequest)
+                for entity in fetchedEntity {
+                    context.delete(entity)
+                }
+                try context.save()
+            }
+        )
+        func createSemester(
+            year: Int,
+            semester: String,
+            gpa: Float,
+            earnedCredit: Float,
+            semesterRank: Int,
+            semesterStudentCount: Int,
+            overallRank: Int,
+            overallStudentCount: Int,
+            lectures: [LectureDetail]?,
+            in context: NSManagedObjectContext
+        ) {
+            let semesterEntity = CDSemester(context: context)
+            semesterEntity.year = Int16(year)
+            semesterEntity.semester = semester
+            semesterEntity.gpa = gpa
+            semesterEntity.earnedCredit = earnedCredit
+            semesterEntity.semesterRank = Int16(semesterRank)
+            semesterEntity.semesterStudentCount = Int16(semesterStudentCount)
+            semesterEntity.overallRank = Int16(overallRank)
+            semesterEntity.overallStudentCount = Int16(overallStudentCount)
+            
+            let lectureEntities = lectures?.compactMap { lecture -> CDLecture? in
+                let cdLecture = CDLecture(context: context)
+                cdLecture.code = lecture.code
+                cdLecture.title = lecture.title
+                cdLecture.credit = Float(lecture.credit)
+                cdLecture.score = lecture.score
+                cdLecture.grade = lecture.grade.rawValue
+                cdLecture.professorName = lecture.professorName
+                return cdLecture
+            }
+            lectureEntities?.forEach { semesterEntity.addToLectures($0) }
+        }
+    }()
+    
+    static let previewVale: GradeClient = GradeClient {
```

Comment:
```suggestion
    static let previewValue: GradeClient = GradeClient {
```
여기 오타가 있어서 previewValue가 적용이 안되는 것 같아요!